src/main/java/it/reply/orchestrator/service/deployment/providers/ChronosServiceImpl.java

```diff
@@ -247,7 +249,16 @@ public boolean isDeployed(DeploymentMessage deploymentMessage) {
 
   @Override
   public void cleanFailedDeploy(DeploymentMessage deploymentMessage) {
-    // DO NOTHING
+    doUndeploy(deploymentMessage);
+  }
+
+  @Override
+  public void doProviderTimeout(DeploymentMessage deploymentMessage) {
+    if (!deploymentMessage.isPollComplete()) {
```

Comment:
If we get here method must throw, so no `if` is needed

Reply:
alternatively there should be a flow going to the finalize deployment task. Use a exclusive gateway in that case (the one with the X)

Reply:
You must try. In the previous configuration, when the timer was not on the boundary, control was necessary because the trigger, once fired, remained in that state even repeating the cycle with a new provider. The component with the 'X' had been installed and behaved like an OR rather than an AND for which the timer trigger operated the exception in any case. I can try to take it off and try it out.

Reply:
done